src/atomate2/forcefields/flows/gruneisen.py

```diff
@@ -0,0 +1,130 @@
+        "seekpath" and "hinuma" are the same definition but
+        seekpath can be used with any kind of unit cell as
+        it relies on phonopy to handle the relationship
+        to the primitive cell and not pymatgen
+    mesh: tuple|float
+        Mesh numbers along a, b, c axes used for Grueneisen parameter computation.
+        Or float to indicate a kpoint density.
+    phonon_displacement_maker: .ForceFieldStaticMaker | None
+    phonon_maker_kwargs: dict
+    perc_vol: float
+        Percent volume to shrink and expand ground state structure
+    compute_gruneisen_param_kwargs: dict
+        Keyword arguments passed to :obj:`compute_gruneisen_param`.
+    """
+
+    name: str = "Gruneisen"
+    bulk_relax_maker: ForceFieldRelaxMaker | None = field(
+        default_factory=lambda: CHGNetRelaxMaker(relax_kwargs={"fmax": 0.00001})
+    )
+    code: str = "forcefields"
+    const_vol_relax_maker: ForceFieldRelaxMaker = field(
+        default_factory=lambda: CHGNetRelaxMaker(
+            relax_kwargs={"fmax": 0.00001}, relax_cell=False
+        )
+    )
+    kpath_scheme: str = "seekpath"
+    phonon_displacement_maker: ForceFieldStaticMaker | None = field(
+        default_factory=CHGNetStaticMaker
+    )
+    phonon_maker_kwargs: dict = field(default_factory=dict)
+    perc_vol: float = 0.01
+    mesh: tuple | float = 7_000
+    compute_gruneisen_param_kwargs: dict = field(default_factory=dict)
+    symprec: float = 1e-4
+
+    @property
+    def prev_calc_dir_argname(self) -> None:
+        """Name of argument informing static maker of previous calculation directory.
+
+        As this differs between different DFT codes (e.g., VASP, CP2K), it
+        has been left as a property to be implemented by the inheriting class.
+
+        Note: this is only applicable if a relax_maker is specified; i.e., two
+        calculations are performed for each ordering (relax -> static)
+        """
+        return
+
+    def initialize_phonon_maker(
+        self,
+        phonon_displacement_maker: ForceFieldStaticMaker,
+        phonon_static_maker: ForceFieldStaticMaker,
+        bulk_relax_maker: ForceFieldRelaxMaker,
+        phonon_maker_kwargs: dict,
+        symprec: float = 1e-4,
+    ) -> PhononMaker | None:
+        """Initialize Phonon Maker.
+
+        Parameters
+        ----------
+        phonon_displacement_maker: .ForceFieldStaticMaker|None
+            Computes Forces for displaced structures in
+            harmonic phonon runs
+        phonon_static_maker: .ForceFieldStaticMaker|None
+            Additional static maker to compute
+            energies and volume after optimization
+        bulk_relax_maker: .ForceFieldRelaxMaker|None
+            Relax Maker for Phonon Run. Typically None.
+        phonon_maker_kwargs: dict
+            Dict to set additional info for phonons.
+
+        Returns
+        -------
+        .PhononMaker
+        """
+        return PhononMaker(
```

Comment:
I'm confused about the logic of this part of the code. Why can't the `BaseGruneisenMaker` just have a `phonon_maker` attribute directly which is overriden by each of the sub-implementations? That would also reduce the total number of attributes the superclass has.

Reply:
Yes, it could, but it might also lead to issues if the 'bulk_relax_maker' is accidentally added. If it's added without changing ISIF to 2 in the INCAR, the entire computation would be lost. I actually made this mistake while modifying the workflow. The current implementation prevents this. I'm also considering creating a dedicated Maker for constant volume optimizations to prevent the user from allowing a volume change...

Reply:
If we made a ConstantVolumeRelaxMaker I think that would be used in a few workflows. So I'm in favour of doing that and then simplying this  part of the code along the lines of my first comment.

Reply:
@utf  i will work on this constant volume optimizert.

I am not sure I am getting the part on the phononmaker. I would prefer not to write a PhononMaker without bulk relaxation or is this what you are suggesting?

Reply:
Just to clarify: the phonon maker currently is used without any optimization. The constant volume relaxation is performed before the phonon run with a separate maker. I would find it very confusing to add this to the phonon maker rather than having a separate maker for the optimization. This is why I added this initialization.

Reply:
I added a constat volume RelaxMaker and TightRelaxMaker. 

(Side note: can we think about how to make writing these Makers, Sets easier in the future? I feel this is still quite complex to write)

Reply:
I'd be happy to chat about making those easier. Would be useful to know which parts are the most bothersome at the moment.

Thanks for explaining more about the phonon maker. I think I misunderstood before. I think the best fix in this case is just to add some code in the `__post_init__` method that checks if the phonon maker has bulk_relax_maker set. It should just print a warning in that case (someone may have a reason for setting it in the future).

E.g., 

```python
@dataclass
class BaseGruneisenMaker(Maker, ABC):
        
        def __post_init__(self):
            if self.phonon_maker.bulk_relax_maker is None:
                 warnings.warn("...")
```

I think this approach will fix the issue you're worried about but also clean up the code of the generalised maker and make doing specific sub implementations easier.

Reply:
> I'd be happy to chat about making those easier. Would be useful to know which parts are the most bothersome at the moment.

There are still a lot of layers in the implementation. Especially also writing this for a new "DFT" code seems to be a lot of effort. Can one maybe unify this a bit more or maybe start from a function to write inputs? You remember that I tried to reimplement this for Quantum Espresso (just for showing it to new users): https://github.com/materialdigital/ADIS2023/blob/main/jobflow.ipynb 

> Thanks for explaining more about the phonon maker. I think I misunderstood before. I think the best fix in this case is just to add some code in the `__post_init__` method that checks if the phonon maker has bulk_relax_maker set. It should just print a warning in that case (someone may have a reason for setting it in the future).
> 
> E.g.,
> 
> ```python
> @dataclass
> class BaseGruneisenMaker(Maker, ABC):
>         
>     def __post_init__(self):
>         if self.phonon_maker.bulk_relax_maker is None:
>             warnings.warn("...")
> ```
> 
> I think this approach will fix the issue you're worried about but also clean up the code of the generalised maker and make doing specific sub implementations easier.

I will use this then instead.

Reply:
@utf Yeah, I tried this. However, there are at least three points that can go wrong. Not sure, the updated implementation now is better.